pkg/controller/controller.go

```diff
@@ -172,10 +177,27 @@ func supportsPluginControllerService(conn *grpc.ClientConn, timeout time.Duratio
 
 	rsp, err := client.GetPluginCapabilities(ctx, &req)
 	if err != nil {
-		return false, err
+		return nil, err
+	}
+	return rsp.GetCapabilities(), nil
+}
+
+func getControllerCapabilities(conn *grpc.ClientConn, timeout time.Duration) ([]*csi.ControllerServiceCapability, error){
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	client := csi.NewControllerClient(conn)
+	req := csi.ControllerGetCapabilitiesRequest{}
+
+	rsp, err := client.ControllerGetCapabilities(ctx, &req)
+	if err != nil {
+		return nil, err
 	}
-	caps := rsp.GetCapabilities()
-	for _, cap := range caps {
+	return rsp.GetCapabilities(), nil
+}
+
+func supportsPluginControllerService(capabilities []*csi.PluginCapability) bool {
```

Comment:
You could have one helper method that searches capabilities for the passed in service